web/src/components/tasks/NaicsCodeTask.test.tsx

```diff
@@ -412,19 +412,59 @@ describe("<NaicsCodeTask />", () => {
       });
     });
 
-    it("prepends register to the next button", async () => {
+    it("opens Needs Account modal on NAICS Code radio button click", () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
       renderPage();
-      expect(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByTestId("naics-radio-621399"));
+
+      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
```

Comment:
[boulder] I think it'd be valuable to also check that the radio buttons are not actually checked. (or on the input box, that there is no value typed into it)

Imagine someone is moving some code around, or say - refactoring the existing logic to use a guard clause:

```
if (isAuthenticated !== IsAuthenticated.FALSE) {
  setShowNeedsAccountModal(true);
} 

setNaicsCode(value);
      if (value.length === LENGTH) {
        ... and so on
```

What's wrong with this code? At first glance, it seems okay. And more importantly, all these tests would pass using this code.

But the code is incorrect - it's missing the `return` in the guard clause, which would mean that it would execute BOTH the open modal AND the rest of the function to save the input value.

The tests should be written in a way that describes the complete behavior so that writing it in this way would cause a failing test, indicating the problem.

This can be done by adding an additional expect statement that expects the input value to be empty, or that the radio input HTML element is not `checked`

Reply:
It looks like this has been added on `line 430` - solid add!